CMakeLists.txt

```diff
@@ -16,51 +16,29 @@ enable_testing()
 set (LUA_PATH "LUA_PATH=${PROJECT_SOURCE_DIR}/?.lua\\;${PROJECT_SOURCE_DIR}/?/init.lua\\;\\;")
 set (LUA_SOURCE_DIR "LUA_SOURCE_DIR=${PROJECT_SOURCE_DIR}")
 
-## VERSION ####################################################################
-###############################################################################
-
-if (NOT VERSION)
-  execute_process(
-    COMMAND git describe --tags --always
-    WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
-    OUTPUT_STRIP_TRAILING_WHITESPACE
-    OUTPUT_VARIABLE VERSION
-    ERROR_QUIET
-  )
-endif()
-
-if (VERSION)
-  configure_file (
-    "${PROJECT_SOURCE_DIR}/${PROJECT_NAME}/VERSION.lua.in"
-    "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}/VERSION.lua"
-  )
-elseif(NOT LUAROCKS) # allow luarocks to build without passing VERSION
-  message(FATAL_ERROR "VERSION is not provided")
-endif()
-
 ## Install ####################################################################
 ###############################################################################
 
-install(
-  FILES ${CMAKE_CURRENT_SOURCE_DIR}/cartridge-cli.lua
-  DESTINATION ${TARANTOOL_INSTALL_LUADIR}/
-)
-
-install(
-  DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/templates
-            ${CMAKE_CURRENT_SOURCE_DIR}/${PROJECT_NAME}
-  USE_SOURCE_PERMISSIONS
-  DESTINATION ${TARANTOOL_INSTALL_LUADIR}/
-  PATTERN "*.in" EXCLUDE
+# Create tempdir
+execute_process(
+  COMMAND mktemp -d
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+  OUTPUT_VARIABLE TMPDIR
 )
 
-install(
-  DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}
-  DESTINATION ${TARANTOOL_INSTALL_LUADIR}/
-  OPTIONAL
+# Generate cartridge executable in tmpdir
+execute_process(
+  COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/compile_executable.lua
```

Comment:
You have a `findTarantool.cmake` script in `cmake` directory. I think you should find tarantool and run this script using it.

Reply:
It's already called https://github.com/tarantool/cartridge-cli/pull/180/files#diff-af3b638bc2a3e6c650974192a53c7291L12

Reply:
I mean you should use `COMMAND ${TARANTOOL_INSTALL_BINDIR}/tarantool ${CMAKE_CURRENT_SOURCE_DIR}/compile_executable.lua` or smth similar.

Main idea - to specify tarantool binary directly as you've already found it.